### ISSUE TITLE
Ignore simple sum in `no-array-reduce`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1348]
I still think 

```js
foo.reduce((total, element) => total + element.bar, 0);
```

should be preferred over

```js
let total = 0;
for (const element of foo) {
	total += element.bar;
}
```

Especially, where `reduce` is not an expression, like this case

```js
if (foo.reduce((total, element) => total + element.bar, 0) > 100) {};
```

Or at least allow to ignore it.

**Comment by sindresorhus** [comment]
I agree. It makes sense for sum.

**Comment by sindresorhus** [comment]
Accepted.
